packages/core/src/new-api/internal/execution/execution-state-reducer.ts

```diff
@@ -23,21 +26,30 @@ import {
 } from "../types/execution-state";
 import { assertIgnitionInvariant } from "../utils/assertions";
 
-import { isExecutionFailure } from "./guards";
+import {
+  isExecutionFailure,
+  isExecutionHold,
+  isExecutionSuccess,
+  isExecutionTimeout,
+} from "./guards";
 import { onchainActionReducer } from "./onchain-action-reducer";
 
 export function executionStateReducer(
   executionStateMap: ExecutionStateMap,
   action: JournalableMessage
 ): ExecutionStateMap {
+  if (isStartRunMessage(action)) {
+    return setTimeoutFuturesToStarted(executionStateMap);
```

Comment:
On a start run message we update the state of all timed out futures to be `STARTED` instead. Let us process them in the batch.